components/dashboard/src/projects/NewProject.tsx

```diff
@@ -229,54 +259,73 @@ export default function NewProject() {
+                            <h3 className="mb-2 text-gray-600 dark:text-gray-400">
+                                No Access
+                            </h3>
+                            <span className="dark:text-gray-500">
+                                Authorize GitHub (github.com) or select a different account.
+                            </span>
+                            <br />
+                            <button className="mt-6" onClick={() => reconfigure()}>Authorize</button>
+                        </div>
+                    </div>)}
+                </div>
+
+            </div>
+            {isGitHub() && (
+                    <div className="pt-3 bg-gray-100 rounded-b-xl border-gray-100 border">
+                        <div className="text-gray-500 text-center w-96 mx-8">
+                            Repository not found? <a href="javascript:void(0)" onClick={e => reconfigure()} className="text-gray-400 underline underline-thickness-thin underline-offset-small hover:text-gray-600">Reconfigure</a>
+                        </div>
+                        {isGitHub() && noOrgs && (
+                            <div className="text-gray-500 mx-auto text-center">
+                                Missing organizations? <a href="javascript:void(0)" onClick={e => grantReadOrgPermissions()} className="text-gray-400 underline underline-thickness-thin underline-offset-small hover:text-gray-600">Grant permissions</a>
                             </div>
-                        ))}
+                        )}
                     </div>
-                )}
-                {reposToRender.length === 0 && (
-                    <p className="text-center ">not found</p>
-                )}
-            </div>
-
-            <div className="px-3 pt-3 bg-gray-100">
-                <div className="text-gray-500 text-center">
-                    Repository not found? <a href="javascript:void(0)" onClick={e => reconfigure()} className="text-gray-400 underline underline-thickness-thin underline-offset-small hover:text-gray-600">Reconfigure</a>
-                </div>
-                {isGitHub() && noOrgs && (
-                    <div className="text-gray-500 mx-auto text-center">
-                        Missing organizations? <a href="javascript:void(0)" onClick={e => grantReadOrgPermissions()} className="text-gray-400 underline underline-thickness-thin underline-offset-small hover:text-gray-600">Grant permissions</a>
+            )}
+        </>
+        );
+
+        const renderEmptyState = () => (<div>
+            <div className="mt-8 border rounded-xl border-gray-100 dark:border-gray-700 flex-col">
+                <div>
+                    <div className="px-12 py-16 text-center text-gray-500 bg-gray-50 dark:bg-gray-800 rounded-xl">
+                        <h3 className="mb-2 text-gray-600 dark:text-gray-400">
+                            Loading ...
```

Comment:
**issue**: The flow now stops with a _Loading ..._ placeholder when there's no active GitLab connection. Expected? There's also _**no**_ error in console. Should we add a flow step to authorize ad-hoc with GitLab.com as we do with users opening workspaces for repository from non-authorized providers, see https://github.com/gitpod-io/gitpod/issues/4197? Sounds good also as a follow up issue if needed. 🏓

Reply:
reproduce and describe how that happened, please? it's a combination of effect, I suspect. 

the logic is: if gitlab.com is connected, use that, otherwise if github.com is connected, use that, otherwise nothing (unexpected, right?) we can remove this dead end by showing the git provider selection. TODO

Reply:
Agree on the logic here! 🤓 

Steps to reproduce:
1. Login using GitHub at [`/login`](https://at-gitlab-projects.staging.gitpod-dev.com/login).
2. Make sure there's no active integration with GitLab at [`/integrations`](https://at-gitlab-projects.staging.gitpod-dev.com/integrations).
3. Go to Projects at [`/projects`](https://at-gitlab-projects.staging.gitpod-dev.com/projects).
4. Click New Project button on the Projects empty state page.
5. Flow stops with the _Loading ..._ placeholder trying to select a git repository on github.com.

**question**: Oh, maybe this is because the GitHub app is not configured for this preview environment, right?

Reply:
I could reproduce it. Indeed this happens to happen if the GH App is not configured, but it would also be the case when no app installations are found, thus a very relevant case. Next deployment should fix this.

Reply:
Thanks, @AlexTugarev! 🌟 

Works as expected. FYI, switching between providers also includes the delay mentioned in https://github.com/gitpod-io/gitpod/pull/5120#discussion_r689670664 when switching back to GitLab. 🙈